### DISCUSSION TITLE
Grant access to all tables in share

### DISCUSSION THREAD

**Discussion by natashamathur** [Snowflake-Labs/terraform-provider-snowflake#2634] [Q&A]
I've created a share to share out data from our main Snowflake database. We gave the share permission using:

resource "snowflake_grant_privileges_to_share" "xyz_share_grant" {

xyz_share now has access to the database and schema, but not all the tables within it. How do you do the equivalent of:

`grant all on all tables in schema xyz_schema to share xyz_share`

with the new granting patterns?

**Comment by sfc-gh-jcieslak** [comment]
you cannot grant "all" privileges to share, it's not available (see [here](https://docs.snowflake.com/en/sql-reference/sql/grant-privilege-share#syntax)). But with specified privileges I guess it would look something like:
```terraform
resource "snowflake_grant_privileges_to_share" "xyz_share_grant" {
  to_share = "todo"
  privileges = ["todo"]
  on_all_tables_in_schema = "todo"
}
```
